markup_fmt/src/parser.rs

```diff
@@ -2569,6 +2562,18 @@ fn is_tag_name_char(c: char) -> bool {
         || c == '\\'
 }
 
+/// Checks whether a character is valid in an HTML tag name, taking into account
+/// special rules for specific template languages.
+///
+/// For example:
+/// - Astro allows '>' in tag names (for fragments)
+/// - Jinja allows '{' for template expressions like <{{ tag_name }}>
+fn is_tag_name_char_for_language(c: char, language: &Language) -> bool {
+    is_tag_name_char(c)
```

Comment:
The name "is_tag_name_char_for_language" is confusing, and I think this function shouldn't call `is_tag_name_char`, splitting them into two different calls.

Reply:
Maybe we could do the following renames:
- `is_tag_name_char` ->  `is_simple_tag_name_char`
- `is_tag_name_char_for_language` -> `is_tag_name_char` 

Or if you want the functions to be splitted, maybe:
- `is_tag_name_char_for_language` ->  `is_special_tag_name_char`
- `is_tag_name_char` ->  `is_simple_tag_name_char`

I think simple and special is probably explicit enough in the context of markup_fmt ?

Reply:
"simple" can be "html".

Reply:
Way better, thanks @g-plane. 
I did the rename and the split you recommended